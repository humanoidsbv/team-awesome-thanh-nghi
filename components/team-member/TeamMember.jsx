import PropTypes from 'prop-types';
import React from 'react';

import './team-member.scss';

const TeamMember = ({
  currentClient, description, firstName, lastName, memberNumber, startDate
}) => (
  <React.Fragment>
    <ul className="member-content">
      <li className="member-content__item">
        <img
          alt="Person Icon"
          className="member-content__logo"
          src="../static/images/icon-person.svg"
        />
        <div className="member-content__wrapper-name">
          <div className="member-content__element member-content__title">{`${firstName} ${lastName}`}</div>
          <div className="member-content__element member-content__subtitle">{description}</div>
        </div>
        <div className="member-content__wrapper member-content__wrapper--first">
          <div className="member-content__element member-content__title">{memberNumber}</div>
          <div className="member-content__element member-content__subtitle">Employee number</div>
        </div>
        <div className="member-content__wrapper">
          <div className="member-content__element member-content__title">{currentClient}</div>
          <div className="member-content__element member-content__subtitle">Current Employer</div>
        </div>
        <div className="member-content__wrapper">
          <div className="member-content__element member-content__title">{startDate}</div>
          <div className="member-content__element member-content__subtitle">Starting date</div>
        </div>
        <button
          className="member-content__button"
          type="button"
        >
          <img
            alt="Arrow down"
            className="member-content__button-img"
            src="../static/images/icon-arrow-down.svg"
          />
        </button>
      </li>
    </ul>
  </React.Fragment>
);

TeamMember.propTypes = {
  currentClient: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired,
  memberNumber: PropTypes.string.isRequired,
  startDate: PropTypes.string.isRequired
};

export default TeamMember;
