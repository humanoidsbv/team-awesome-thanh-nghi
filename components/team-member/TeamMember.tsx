import React from 'react';

import './team-member.scss';

interface TeamMemberProps {
  currentClient: string;
  description: string;
  firstName: string;
  lastName: string;
  memberNumber: string;
  startDate: string;
}

const TeamMember = ({
  currentClient, description, firstName, lastName, memberNumber, startDate
}: TeamMemberProps) => (
  <React.Fragment>
    <ul className="member-content">
      <li className="member-content__item">
        <img
          alt="Person Icon"
          className="member-content__logo"
          src="../static/images/icon-person.svg"
        />
        <div className="member-content__wrapper-name">
          <div className="member-content__element member-content__title">{`${firstName} ${lastName}`}</div>
          <div className="member-content__element member-content__subtitle">{description}</div>
        </div>
        <div className="member-content__wrapper member-content__wrapper--first">
          <div className="member-content__element member-content__title">{memberNumber}</div>
          <div className="member-content__element member-content__subtitle">Employee number</div>
        </div>
        <div className="member-content__wrapper">
          <div className="member-content__element member-content__title">{currentClient}</div>
          <div className="member-content__element member-content__subtitle">Current Employer</div>
        </div>
        <div className="member-content__wrapper">
          <div className="member-content__element member-content__title">{startDate}</div>
          <div className="member-content__element member-content__subtitle">Starting date</div>
        </div>
        <button
          className="member-content__button"
          type="button"
        >
          <img
            alt="Arrow down"
            className="member-content__button-img"
            src="../static/images/icon-arrow-down.svg"
          />
        </button>
      </li>
    </ul>
  </React.Fragment>
);

export default TeamMember;
