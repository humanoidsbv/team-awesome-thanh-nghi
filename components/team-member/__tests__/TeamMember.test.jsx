import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import TeamMember from '../TeamMember.tsx';

const teamMember = {
  clientName: 'Humanoids',
  currentClient: '3',
  firstName: 'Pietje',
  lastName: 'Puk',
  description: 'Front End Developer',
  memberNumber: 'H-007',
  address: 'Goudseweg 12',
  zipcode: '6857SK',
  city: 'Verweggiestad',
  emailAddress: 'pietje.puk@humanoids.nl',
  linkedIn: 'linkedin.com/pietje-puk',
  startDate: '2019-03-03T15:07:44.742Z',
  id: 'aSMiO9N'
};

it('renders a snapshot', () => {
  const wrapper = shallow(
    <TeamMember
      currentClient={teamMember.clientName}
      description={teamMember.description}
      firstName={teamMember.firstName}
      key={teamMember.id}
      lastName={teamMember.lastName}
      memberNumber={teamMember.memberNumber}
      startDate={teamMember.startDate}
    />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
