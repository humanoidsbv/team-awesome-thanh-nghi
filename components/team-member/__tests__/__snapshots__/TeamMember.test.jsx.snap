// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<Fragment>
  <ul
    className="member-content"
  >
    <li
      className="member-content__item"
    >
      <img
        alt="Person Icon"
        className="member-content__logo"
        src="../static/images/icon-person.svg"
      />
      <div
        className="member-content__wrapper-name"
      >
        <div
          className="member-content__element member-content__title"
        >
          Pietje Puk
        </div>
        <div
          className="member-content__element member-content__subtitle"
        >
          Front End Developer
        </div>
      </div>
      <div
        className="member-content__wrapper member-content__wrapper--first"
      >
        <div
          className="member-content__element member-content__title"
        >
          H-007
        </div>
        <div
          className="member-content__element member-content__subtitle"
        >
          Employee number
        </div>
      </div>
      <div
        className="member-content__wrapper"
      >
        <div
          className="member-content__element member-content__title"
        >
          Humanoids
        </div>
        <div
          className="member-content__element member-content__subtitle"
        >
          Current Employer
        </div>
      </div>
      <div
        className="member-content__wrapper"
      >
        <div
          className="member-content__element member-content__title"
        >
          2019-03-03T15:07:44.742Z
        </div>
        <div
          className="member-content__element member-content__subtitle"
        >
          Starting date
        </div>
      </div>
      <button
        className="member-content__button"
        type="button"
      >
        <img
          alt="Arrow down"
          className="member-content__button-img"
          src="../static/images/icon-arrow-down.svg"
        />
      </button>
    </li>
  </ul>
</Fragment>
`;
